enigma-runtime-t/src/lib.rs

```diff
@@ -381,11 +378,48 @@ impl Runtime {
             }
         }
     }
+
+    pub fn encrypt_with_nonce(&mut self, args: RuntimeArgs) -> Result<()> {
+        let message_ptr: u32 = args.nth_checked(0)?;
+        let message_len: u32 = args.nth_checked(1)?;
+        let message = self.memory.get(message_ptr, message_len as usize)?;
+
+        let key_ptr: u32 = args.nth_checked(2)?;
+        let mut key: SymmetricKey = [0u8; SYMMETRIC_KEY_SIZE];
+        self.memory.get_into(key_ptr, &mut key)?;
+
+        let iv_ptr: u32 = args.nth_checked(3)?;
+        let mut iv: IV = [0u8; IV_SIZE];
+        self.memory.get_into(iv_ptr, &mut iv)?;
+
+        let ptr: u32 = args.nth_checked(4)?;
+        let enc_message = encrypt_with_nonce(&message, &key, Some(iv))
+            .map_err(|err| WasmError::EnclaveError(EnclaveError::SystemError(EnclaveSystemError::CryptoError { err })))?;
+        self.memory.set(ptr, &enc_message)?;
+        Ok(())
+    }
+
+    pub fn decrypt(&mut self, args: RuntimeArgs) -> Result<()> {
+        let cipheriv_ptr: u32 = args.nth_checked(0)?;
+        let cipheriv_len: u32 = args.nth_checked(1)?;
+        let cipheriv = self.memory.get(cipheriv_ptr, cipheriv_len as usize)?;
+
+        let key_ptr: u32 = args.nth_checked(2)?;
+        let mut key: SymmetricKey = [0u8; SYMMETRIC_KEY_SIZE];
+        self.memory.get_into(key_ptr, &mut key)?;
+
+        let ptr: u32 = args.nth_checked(3)?;
+        let message = decrypt(&cipheriv, &key)
+            .map_err(|err| WasmError::EnclaveError(EnclaveError::SystemError(EnclaveSystemError::CryptoError { err })))?;
```

Comment:
Please implement the need From for errors. We did an effort to keep the runtime readable and free from conversions :) See https://github.com/enigmampc/enigma-core/blob/develop/enigma-tools-t/src/common/errors_t.rs for the reference.

Reply:
Done.